File: Praktikum0 Python/PRAK005-2510817220002-AishaZwarintyanda.py
def AZ(n=7, s=3):
    #n & s bisa diganti dgn var lain
    for a in range(n):
        # huruf A
        Aletter = "".join(
            #join fungsinya untuk menggabung byk string, jika tdk ada .join maka coding tdk bisa jalan
            "#" if a == 0 or a == n // 2 or z == 0 or z == n - 1 else " "
            for z in range(n)
        )
        #a == 0/n menentukan posisi garis atas dan tengah A
        #z == 0/n - 1 menentukan posisi garis kanan dan kiri huruf a
        #n//2 maksudnya untuk membulatkan nilai setengah dari n, tapi dibulatkan ke bawah
        #Kalau n = 7 → n // 2 = 3
        #Kalau n = 8 → n // 2 = 4
        #range tidak bisa diganti dengan nama lain karena fungsi bawaan dari python sendiri
        #urutan if or for dan in: if menjalankan semua or, lalu for menunjukan bahwa rumus sebelumnya untuk diterapkan ke "in range(n)"

        # huruf Z
        Zletter = "".join(
            "#" if a == 0 or a == n - 1 or z == n - 1 - a else " "
            for z in range(n)
        )

        print(Aletter + " " * s + Zletter)
        #space "" akan dikalikan berdasar jumlah s terlebih dahulu, baru ditambahkan huruf a dan z diakhir


#call output
AZ(7, 3)
#angka depan menentukan tinggi huruf, angka belakang menentukan lebar space huruf
#misal: jika diisi angka 0, huruf akan berdempetan, diatas 0 akan memberi jarak antar huruf

File: Praktikum0 C/PRAK005-2510817220002-AishaZwarintyanda.c
#include <stdio.h>

int main() {
    int a, z;
    int n = 7; // tinggi huruf
    int spasi = 3; // jarak antar huruf

    for (a = 0; a < n; a++) {
    //huruf a
    for (z = 0; z < n; z++)
    //z = 0 agar sesuai dengan jumlah int, jika kurang huruf akan terpotong
    //z<n adalah proses berhenti loop z pada huruf n(7)
    //huruf a mewakili baris dan z adalah kolum, keduanya akan digunakan di setiap huruf
        
        {
            if (a == 0)
            //membuat bagian atas huruf a
                printf("#");
            else if (a == n / 2)
                printf("#");
            else if (z == 0 || z == n - 1)
            //membuat kaki huruf a di kanan & kiri
                printf("#");
            else
                printf(" ");
        }

        // spasi antar huruf
        for (z = 0; z < spasi; z++) {
            printf(" ");
        }

        //huruf z
        for (z = 0; z < n; z++) {
            if (a == 0 || a == n - 1)
            //membuat kaki huruf z di kanan & kiri
            // jika nilai n - 1 diganti dgn angka lain, akan mempengaruhi letak garis bawah huruf z (misal - 7, garis bawah z akan berada dibawah garis atas)
                printf("#");
            else if (z == n - 1 - a)
                printf("#");
            else
                printf(" ");
            //fungsi if, if else, dan else disini untuk mengatur output mana yang diperlukan
        }
        printf("\n");
        //jika tdk ada break, maka hasil akan dioutputkan secara berjejer
    }

    return 0;
}
